File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/Java_Gradle_Project/Laboratory_1/FestivalTicketSale/src/main/java/org/example/Model/Entity.java
package org.example.Model;

import java.io.Serializable;
import java.util.Objects;

public class Entity<ID> implements Serializable {
    private ID id;
    public ID getId()
    {
        return id;
    }

    public void setId(ID id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Entity<?> entity)) return false;
        return Objects.equals(getId(), entity.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId());
    }
}

File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/C#_Project/Laboratory_8_Proto/FestivalTicketSaleCS/model/Entity.cs
using System;

namespace model
{[Serializable]
    public class Entity<TId>
    {
        public Entity() {}

        public Entity(TId id)
        {
            this.Id = id;
        }

        public TId Id
        {
            get;
            set;
        }
    }
}

